### DISCUSSION TITLE
Quarto - Confluence Publishing

### DISCUSSION THREAD

**Discussion by giscus** [quarto-dev/quarto-cli#11409] [Blog]
# Quarto - Confluence Publishing

Quarto 1.3 adds support for publishing individual documents, and projects containing many documents to Atlassian Confluence.

https://quarto.org/docs/blog/posts/2023-03-20-confluence/

**Comment by statquant** [comment]
Looks cool, does it support sending html documents within confluence html macros as well or just markdown and static pictures?

**Reply by ghost** [reply]
Hello, thanks for posting this question.

Can you please clarify a bit more by "sending html documents"?

We do [expose a Confluence Raw Block](https://github.com/quarto-dev/quarto-cli/issues/4472) that lets you put in arbitrary[ Confluence Storage Format](https://confluence.atlassian.com/doc/confluence-storage-format-790796544.html).  This would let you add CSF macros like Task Lists and Page Trees to your documents.  

I'm not sure if there are other HTML Macros that could be supported but we are interested in learning if there are more ways we can add functionality to Confluence publishing.

**Comment by grepinsight** [comment]
Thank you for this awesome feature @allenmanning . I have been testing it and it works great.

One question/feature request: Can I publish the minimal changes possible? For example, if I have 200 documents already published in Confluence using Quarto and I need to add one more, can I just publish that one document without having to re-render and publish all of them again? 

For example, I have Confluence integration with Slack, and every time I publish, I get notifications for all documents...

**Reply by ghost** [reply]
> Can I just publish that one document without having to re-render and publish all of them again?

Thanks for posting @grepinsight.  This enhancement is now [captured in this issue](https://github.com/quarto-dev/quarto-cli/issues/5698).  

We may be able to save a hash of the qmd content as metadata to short-circuit an wasteful and noisy re-publish.  CSF is converted in deterministic ways once it is sent to the cloud so this may be our best shot.

**Reply by grepinsight** [reply]
Thank you for creating the issue!
